dlc-manager/src/contract/accepted_contract.rs

```diff
@@ -50,6 +50,31 @@ impl AcceptedContract {
 
         string_id
     }
+
+    /// Compute the profit and loss for this contract and an assciated cet index
+    pub fn compute_pnl(&self, cet_index: usize) -> i64 {
+        let offer = &self.offered_contract;
+        let party_params = if offer.is_offer_party {
+            &offer.offer_params
+        } else {
+            &self.accept_params
+        };
+        let collateral = party_params.collateral as i64;
+        let cet = &self.dlc_transactions.cets[cet_index];
+        let v0_witness_payout_script = &party_params.payout_script_pubkey;
+        let final_payout = cet
+            .output
+            .iter()
+            .find_map(|x| {
+                if &x.script_pubkey == v0_witness_payout_script {
+                    Some(x.value)
+                } else {
+                    None
+                }
+            })
+            .unwrap_or(0) as i64;
+        final_payout - collateral
+    }
```

Comment:
I moved this implementation from the sample code into a method for accepted contracts, so that pnl can be computed internally. Perhaps this means unit tests should be written for this method?

Reply:
Unit tests would be great, one thing that might be annoying is constructing the contract, maybe you can use some json input or something to make it easier.